assessment/models.py

```diff
@@ -182,14 +182,10 @@ class Meta:
     def __str__(self):
         return str(self.pk)
 
-    def save(self, *args, **kwargs):
+    def clean(self, *args, **kwargs):
         if self.survey.is_private:
-            if self.user == self.survey.admin or self.user in self.survey.users.all():
-                super(Result, self).save(*args, **kwargs)
-            else:
+            if not (self.user == self.survey.admin or self.user in self.survey.users.all()):
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=python%3AS1066)